daemon/endpoint.go

```diff
@@ -141,6 +142,11 @@ func (d *Daemon) createEndpoint(epTemplate *models.EndpointChangeRequest, id str
 		return PutEndpointIDInvalidCode, err
 	}
 	ep.SetDefaultOpts(option.Config.Opts)
+	if k8s.IsEnabled() {
```

Comment:
Do this assignment inside `NewEndpointFromChangeModel`

Reply:
This is resolved by no longer having the Annotator stored within the Endpoint structure.